### PR TITLE
[compact-3] Prepare for Compact - Simplify Default vs Constant Serializers [API-971]

### PR DESCRIPTION
**Pull request by zpqrtbnk**
For some reasons the set of built-in serializers is split between "constant" and "default" serializers though they practically all are constant serializers. This PR merges them all into one namespace and one definition so it simplifies the code in various places.

### CODE REVIEW

File: src/Hazelcast.Net/Serialization/ConstantSerializers/ConstantSerializerDefinitions.cs [hunk 1]
```diff
@@ -43,6 +51,29 @@ public void AddSerializers(SerializationService service)
 
             service.RegisterConstantSerializer<Guid>(new GuidSerializer());
             service.RegisterConstantSerializer<KeyValuePair<object, object>>(new KeyValuePairSerializer());
+
+            //TODO: proper support for generic types
```
Comment (emreyigit) [inline src/Hazelcast.Net/Serialization/ConstantSerializers/ConstantSerializerDefinitions.cs:55]: What does it mean proper support?

**Comment by emreyigit** [comment]
Looks nice. I've just one comment.

File: src/Hazelcast.Net/Serialization/ConstantSerializers/ConstantSerializerDefinitions.cs (hunk 1 above)
Comment (zpqrtbnk) [reply src/Hazelcast.Net/Serialization/ConstantSerializers/ConstantSerializerDefinitions.cs:55]: Interestingly, I have no idea, this TODO and the one a few lines below were copied over from the very original project (v3?) when I joined HZ. I don't understand them either, so I am going to update this PR to remove them entirely.

Comment (zpqrtbnk) [reply src/Hazelcast.Net/Serialization/ConstantSerializers/ConstantSerializerDefinitions.cs]: Have pushed the change.

**Review by emreyigit — DISMISSED** [review]

**Comment by zpqrtbnk** [comment]
Rebased the PR on top of `master` which only has some GitHub workflows updates, so not going to bother anyone with re-reviewing + waiting for tests. Bypassing protection and merging.

**Pull request merged**